### ISSUE TITLE
Opaque function-by-method does not reveal itself

### ISSUE THREAD

**Issue by random1664795529** [dafny-lang/dafny#4536]
### Dafny version

4.0.0

### Code to produce this issue

```dafny
opaque function thisDoesNotWork() : int { 
  4
} by method {
  return 4;
}

opaque function iHaveToDoThis() : int { 
  4
} by method {
  reveal iHaveToDoThis();
  return 4;
}
```

### What happened?

The function `thisDoesNotWork` does not verify as the function is not automatically revealed in the `by-method` part.

It seems very unlikely that we ever want this i.e. the current behavior is bad (a bug). 

If you really wanted this then you could still do something like the below to avoid the function being auto-revealed.

```
opaque function fooSpec...

function foo(...) {
   fooSpec(..)
} by method { .. }
```

### What type of operating system are you experiencing the problem on?

Mac

**Comment by keyboardDrummer** [comment]
I agree the current behavior does not make sense. I'm hoping to do more rework on opaque, and then I'll run into this as well, so I wont't plan to resolve this as a separate bug.
